### ISSUE TITLE
Support user-defined hyper-param schedules

### ISSUE THREAD

**Issue by shane-huang** [intel-analytics/BigDL#354]
Need to support user defined hyper-param (e.g. learning rate) schedule and weight initializers, etc.

**Comment by shane-huang** [comment]
narrow down the scope of this issue and open a new issue for supporting user-defined weight/bias initializers.

**Comment by yiheng** [comment]
weight initializer has been supported.
